tests/convert_valid_utf32_to_latin1_tests.cpp

```diff
@@ -0,0 +1,42 @@
+#include "simdutf.h"
+
+#include <array>
+#include <iostream>
+
+#include <tests/helpers/transcode_test_base.h>
+#include <tests/helpers/random_int.h>
+#include <tests/helpers/test.h>
+#include <memory>
+
+namespace {
+  std::array<size_t, 7> input_size{7, 16, 12, 64, 67, 128, 256};
+
+  using simdutf::tests::helpers::transcode_utf8_to_utf16_test_base;
+
+  constexpr size_t trials = 10000;
+}
+
+TEST(convert_latin1_only) {
+  size_t counter = 0;
+  auto generator = [&counter]() -> uint32_t {
+    return counter++ & 0xFF; // 0x7f;
+  };
+
+  auto procedure = [&implementation](const char32_t* utf32, size_t size, char* latin1) -> size_t {
+    return implementation.convert_valid_utf32_to_latin1(utf32, size, latin1);
+  };
+  auto size_procedure = [&implementation](const char32_t* utf32, size_t size) -> size_t {
+    return implementation.latin1_length_from_utf32(size);
+  };
+  std::array<size_t, 4> input_size{7,16,24,67};
+  for (size_t size: input_size) {
```

Comment:
So here you appear to be iterating through the possible sizes, but you only ever test with size 256, so this is probably wrong?